sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryServicesImpl.java

```diff
@@ -893,6 +903,20 @@ public void deleteDataset(String projectId, String datasetId)
           ignoreInsertIds);
     }
 
+    private GoogleJsonError.ErrorInfo getErrorInfo(IOException e) {
+      GoogleJsonResponseException jsonCause = null;
+      Throwable eCause = e;
+      if (eCause instanceof GoogleJsonResponseException) {
```

Comment:
How do these exceptions extend both GoogleJsonResponseException and IOException ?

Reply:
GoogleJsonResponseException extends HttpResponseException and HttpResponseException extends IOException

Reply:
Thanks. In that case let's simplify this to following.

if (!(e instanceof GoogleJsonResponseException)) {
  return null;
}

GoogleJsonResponseException jsonCause = (GoogleJsonResponseException) e;
...

Reply:
Yup, that should work better.